internal/pkg/cli/local_run.go

```diff
@@ -106,11 +201,182 @@ func (o *localRunOpts) Execute() error {
+
+			containerPort := hostPort
+			if portMapping.ContainerPort == nil {
+				containerPort = strconv.FormatInt(aws.Int64Value(portMapping.ContainerPort), 10)
+			}
+			containerPorts[hostPort] = containerPort
+		}
+	}
+
+	envSess, err := o.sessProvider.FromRole(o.targetEnv.ManagerRoleARN, o.targetEnv.Region)
+	if err != nil {
+		return fmt.Errorf("get env session: %w", err)
+	}
+	mft, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		ws:           o.ws,
+		unmarshal:    o.unmarshal,
+		sess:         envSess,
+	})
+	if err != nil {
+		return err
+	}
+	o.appliedDynamicMft = mft
+	o.repository, err = o.configureClients(o)
+	if err != nil {
+		return err
+	}
+
+	if err := o.buildContainerImages(o); err != nil {
+		return err
+	}
+
+	for name, imageInfo := range o.out.ImageDigests {
+		o.imageInfoList = append(o.imageInfoList, clideploy.ImagePerContainer{
+			ContainerName: name,
+			ImageURI:      imageInfo.ImageName,
+		})
+	}
+
+	var sidecarImageLocations []clideploy.ImagePerContainer //sidecarImageLocations has the image locations which are already built
+	manifestContent := o.appliedDynamicMft.Manifest()
+	switch t := manifestContent.(type) {
+	case *manifest.ScheduledJob:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	case *manifest.LoadBalancedWebService:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	case *manifest.WorkerService:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	case *manifest.BackendService:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	}
+	o.imageInfoList = append(o.imageInfoList, sidecarImageLocations...)
+
+	err = o.runPauseContainer(context.Background(), containerPorts)
+	if err != nil {
+		return err
+	}
+
+	err = o.runContainers(context.Background(), o.imageInfoList, secretsList, envVars)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *localRunOpts) getContainerSuffix() string {
+	return fmt.Sprintf("%s-%s-%s", o.appName, o.envName, o.wkldName)
+}
+
+func getBuiltSidecarImageLocations(sidecars map[string]*manifest.SidecarConfig) []clideploy.ImagePerContainer {
+	//get the image location for the sidecars which are already built and are in a registry
+	var sideCarBuiltImageLocations []clideploy.ImagePerContainer
+	for sideCarName, sidecar := range sidecars {
+		if uri, hasLocation := sidecar.ImageURI(); hasLocation {
+			sideCarBuiltImageLocations = append(sideCarBuiltImageLocations, clideploy.ImagePerContainer{
+				ContainerName: sideCarName,
+				ImageURI:      uri,
+			})
+		}
+	}
+	return sideCarBuiltImageLocations
+}
+
+func (o *localRunOpts) runPauseContainer(ctx context.Context, containerPorts map[string]string) error {
+	containerNameWithSuffix := fmt.Sprintf("%s-%s", pauseContainerName, o.containerSuffix)
```

Comment:
i think we actually want this to be a prefix (same for `runContainers` below) - it should be something like "appName-envName-svcName-pause"